polyfill/lib/yearmonth.mjs

```diff
@@ -1,126 +1,113 @@
+    let months = two.month - one.month;
+    if (months < 0) {
+      years -= 1;
+      months += 12;
+    }
+    const Duration = ES.GetIntrinsic('%Temporal.Duration%');
+    return new Duration(years, months);
+  }
+  toString() {
+    let year = ES.ISOYearString(GetSlot(this, YEAR));
+    let month = ES.ISODateTimePartString(GetSlot(this, MONTH));
+    let resultString = `${year}-${month}`;
+    return resultString;
+  }
+  toLocaleString(...args) {
+    return new Intl.DateTimeFormat(...args).format(this);
+  }
+  withDay(day, disambiguation = 'constrain') {
+    const year = GetSlot(this, YEAR);
+    const month = GetSlot(this, MONTH);
+    const Date = ES.GetIntrinsic('%Temporal.Date%');
+    return new Date(year, month, day, disambiguation);
   }
-  const Duration = ES.GetIntrinsic('%Temporal.Duration%');
-  return new Duration(years, months);
-};
-YearMonth.prototype.toString = YearMonth.prototype.toJSON = function toString() {
-  let year = ES.ISOYearString(GetSlot(this, YEAR));
-  let month = ES.ISODateTimePartString(GetSlot(this, MONTH));
-  let resultString = `${year}-${month}`;
-  return resultString;
-};
-YearMonth.prototype.toLocaleString = function toLocaleString(...args) {
-  return new Intl.DateTimeFormat(...args).format(this);
-};
-YearMonth.prototype.withDay = function withDay(day, disambiguation = 'constrain') {
-  const year = GetSlot(this, YEAR);
-  const month = GetSlot(this, MONTH);
-  const Date = ES.GetIntrinsic('%Temporal.Date%');
-  return new Date(year, month, day, disambiguation);
-};
-
-YearMonth.fromString = function fromString(isoStringParam) {
-  const isoString = ES.ToString(isoStringParam);
-  const match = DATE.exec(isoString);
-  if (!match) throw new RangeError('invalid date string');
-  const year = ES.ToInteger(match[1]);
-  const month = ES.ToInteger(match[2]);
-  return new YearMonth(year, month, 'reject');
-};
-YearMonth.compare = function compare(one, two) {
-  if (one.year !== two.year) return one.year - two.year;
-  if (one.month !== two.month) return one.month - two.month;
-  return 0;
-};
 
-Object.defineProperty(YearMonth.prototype, Symbol.toStringTag, {
-  value: 'Temporal.YearMonth'
-});
+  static fromString(isoStringParam) {
+    isoString = ES.ToString(isoString);
+    const match = STRING.exec(isoString);
+    if (!match) throw new RangeError(`invalid yearmonth: ${isoString}`);
+    const year = ES.ToInteger(match[1]);
+    const month = ES.ToInteger(match[2]);
+    return new ES.GetIntrinsic('%Temporal.YearMonth%')(year, month, 'reject');
+  }
+  static from(...args) {
+    return ES.GetIntrinsic('%Temporal.yearmonth%')(...args);
+  }
+  static compare(one, two) {
+    one = ES.GetIntrinsic('%Temporal.yearmonth%')(one);
+    two = ES.GetIntrinsic('%Temporal.yearmonth%')(two);
+    if (one.year !== two.year) return one.year - two.year;
+    if (one.month !== two.month) return one.month - two.month;
+    return 0;
+  }
+}
+YearMonth.prototype.toJSON = YearMonth.prototype.toString;
```

Comment:
This should probably be a nonenumerable property

Reply:
They are non-enumerable on instances since they are on the prototype.

Just had a look at other built-ins: none of them and none of their prototypes have ANY enumerable properties. So I'll go with the same thing and make all properties on the class & the prototype non-enumerable.